src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -1859,7 +1859,7 @@ boolean isSerializable() {
                 if (forceSerializable) {
                     return true;
                 }
-                return types.asSuper(tree.target, syms.serializableType.tsym) != null;
+                return types.isSubtype(tree.target, syms.serializableType);
```

Comment:
That's another "is this a subclass of XYZ" check - asSuper is fine here (with ref projection).

Reply:
OK